Examples/test-suite/special_variable_macros.i

```diff
@@ -39,10 +39,13 @@ struct NameWrap {
 private:
   Name name;
 };
+
+// Global variable for testing whether examplekw was touched
+int accessed_examplekw = 0;
 %}
 
 // check $1 and $input get expanded properly when used from $typemap()
-%typemap(in) Name *GENERIC ($*1_type temp)
+%typemap(in, examplekw="accessed_examplekw=1;") Name *GENERIC ($*1_type temp)
```

Comment:
Hm... I thought it might be OK behavior if the keyword arguments could also use the local variables: but alas, with:
```
%typemap(in, examplekw="accessed_examplekw=1; (void)temp;") Name *GENERIC ($*1_type temp)
```
the `temp` inside `examplekw` is not correctly substituted.

I'm still trying to debug what's going on in the bowels of `replace_embedded_typemap` but hope to have a fix soon...

Reply:
OK @wsfulton I think I have fixed this bug. I've also checked the behavior with the Fortran test suite which uses the typemap-keyword feature more extensively.

Reply:
I see the temporary variables as being associated with the typemap body, not any attributes.